.buildkite/e2e/pipeline-gen/pipeline.tpl.yaml

```
@@ -29,9 +28,9 @@ steps:
     agents:
       {{- if $test.Dind }}
       provider: "gcp"
-      image: "family/core-ubuntu-2004"
-      {{- if eq $test.Provider "kind" }}
-      machineType: "{{ .KindAgentsMachineType }}"
+      image: "family/core-ubuntu-2204"
+      {{- if or (eq $test.Provider "kind") (eq $test.Provider "k3d") }}
+      machineType: "{{ $.KindAgentsMachineType }}"
```

```suggestion
      machineType: "{{ $.DockerInDockerMachineType }}"
```
or `K8sInDockerMachineType` as this is no longer just used for Kind.